src/main/java/at/hannibal2/skyhanni/utils/ComputerTimeOffset.kt

```diff
@@ -0,0 +1,124 @@
+import kotlin.time.Duration.Companion.seconds
+
+@SkyHanniModule
+object ComputerTimeOffset {
+    var offsetMillis: Duration? = null
+        private set
+
+    private var lastCheckTime = SimpleTimeMark.farPast()
+
+    private val config get() = SkyHanniMod.feature.misc.warnAboutPcTimeOffset
+
+    private val offsetFixLinks by lazy {
+        when {
+            OSUtils.isWindows ->
+                @Suppress("ktlint:standard:max-line-length")
+                "https://support.microsoft.com/en-us/windows/how-to-set-your-time-and-time-zone-dfaa7122-479f-5b98-2a7b-fa0b6e01b261"
+            OSUtils.isLinux -> "https://unix.stackexchange.com/a/79116"
+            OSUtils.isMac -> "https://support.apple.com/guide/mac-help/set-the-date-and-time-automatically-mchlp2996/mac"
+            else -> null
+        }
+    }
+
+    init {
+        checkOffset()
+        startMonitoringTimeChanges()
+    }
+
+    private fun checkOffset() {
+        SkyHanniMod.coroutineScope.launch {
+            offsetMillis = getNtpOffset("time.google.com")
+            offsetMillis?.let {
+                println("SkyHanni detected a time offset of $it.")
+            } ?: println("SkyHanni failed to detect a time offset.")
+        }
+    }
+
+    private fun getNtpOffset(ntpServer: String): Duration? = try {
+        val client = NTPUDPClient()
+        val address = InetAddress.getByName(ntpServer)
+        val timeInfo = client.getTime(address)
+
+        timeInfo.computeDetails()
+        timeInfo.offset.milliseconds
+    } catch (e: Exception) {
+        if (LorenzUtils.inSkyBlock && config) ErrorManager.logErrorWithData(e, "Failed to get NTP offset")
+        else e.printStackTrace()
+        null
+    }
+
+    private var lastSystemTime = System.currentTimeMillis()
+    private var lastDetectedOffset = 1.seconds
+
+    private fun startMonitoringTimeChanges() {
+        thread {
+            while (true) {
+                Thread.sleep(1000)
+                detectTimeChange()
+            }
+        }
+    }
+
+    private fun detectTimeChange() {
+        val currentSystemTime = System.currentTimeMillis()
+        val timeDifference = (currentSystemTime - lastSystemTime).milliseconds
+        lastSystemTime = currentSystemTime
+
+        val expectedDuration = 1.seconds
+        val deviation = timeDifference - expectedDuration
+
+        if (deviation.absoluteValue > 1.seconds && config) {
+            ChatUtils.chat("Time Offset changed from ${lastDetectedOffset.format()} to ${deviation.format()}")
+        }
+        lastDetectedOffset = deviation
+    }
+
+    @SubscribeEvent
+    fun onProfileJoin(event: ProfileJoinEvent) {
+        if (!config) return
+        val offsetMillis = offsetMillis ?: return
+        if (offsetMillis.absoluteValue < 5.seconds) return
+
+        ChatUtils.clickableLinkChat(
+            "Your computer's clock is off by ${offsetMillis.format()}. Please update your time settings. Click here for instructions.",
```

Comment:
This here printed SkyHanni detected a time offset of -(3m 55.701s). When my clock is ahead by about 4 mins
![image](https://github.com/user-attachments/assets/40a5124e-ef02-4112-821d-ca1776661b0c)

While the way to find the proper time is SystemTime + offset, the value displayed to the user should either be flipped or just the absolute value